TheTharz/TutorHub
# Feature Addition and UI Enhancements

# Pull Request

---

## Changes Made:

### 1. Added SignUp Screen:
   - Created a new screen `SignUpScreen` with a form for user registration.
   - Integrated a sign-up form with user inputs for name, email, password, and confirmation.
   - Implemented logic to handle form validation, ensuring that required fields are filled and passwords meet length requirements.
   - Provided a checkbox for agreeing to Terms and Conditions.
   - Integrated a Sign-Up button that triggers the `signUp` method in the `SignUpController`.

### 2. Added SignUpController:
   - Created a `SignUpController` class responsible for handling the sign-up logic.
   - Integrated text editing controllers for name, email, password, and confirmation.
   - Implemented a `signUp` method that performs validation, checks internet connectivity, and simulates a loading screen using `TFullScreenLoader`.
   - Utilized GetX state management to manage the state of the sign-up process.

### 3. Refactored FullScreenLoader Usage:
   - Modified the `TFullScreenLoader` class to utilize `WillPopScope` for preventing the dismissal of the loading dialog with the back button.
   - Adjusted the `openLoadingDialog` method to wrap the dialog content with `WillPopScope` for better user experience.

### 4. Adjusted UI Theming:
   - Integrated various UI theming changes based on the specified themes such as app bar theme, button themes, checkbox themes, and more.

### 5. Updated App Theme Configuration:
   - Incorporated the `TAppTheme` class for both light and dark themes.
   - Utilized the provided theme classes for consistent and visually appealing UI across the application.

### 6. Dependency Management:
   - Updated dependencies to the latest versions for improved compatibility and security.
   - Included new dependencies such as `lottie`, `connectivity_plus`, `flutter_native_splash`, and `smooth_page_indicator` for enhanced features.

### 7. Miscellaneous:
   - Structured the project files and organized the code for better readability and maintainability.

## How to Test:

1. Navigate to the `SignUpScreen`.
2. Fill in the required details in the sign-up form.
3. Check the checkbox to agree to the Terms and Conditions.
4. Click on the "Sign Up" button.
5. Observe the loading screen while the sign-up process is simulated.

---

**Reviewer Notes:**

- Please ensure that the dependencies are properly resolved by running `flutter pub get`.
- Test the sign-up flow thoroughly to check for any issues or unexpected behavior.
- Verify the theming changes across different screens to maintain a consistent UI.

---

**Checklist:**

- [x] Created SignUpScreen with a user registration form.
- [x] Implemented SignUpController for sign-up logic and state management.
- [x] Updated FullScreenLoader usage for improved user experience.
- [x] Applied UI theming changes for a visually appealing look.
- [x] Checked and updated dependencies to the latest versions.
- [x] Organized project structure for better readability and maintainability.

---
